### ISSUE TITLE
Initialization fails in absence of MQTT broker

### ISSUE THREAD

**Issue by plundeen** [sabuhish/fastapi-mqtt#45]
### Problem Statement

The MQTT connection gracefully handles reconnects if the connection is lost at runtime, but if it is unable to establish the initial connection on startup, it hard fails.

```text
2023-01-08 12:41:40,616 uvicorn.error  INFO  Started server process [31368]
2023-01-08 12:41:40,617 uvicorn.error  INFO  Waiting for application startup.
2023-01-08 12:41:40,617 uvicorn.error  WARNI Used broker version is 5
2023-01-08 12:41:44,708 uvicorn.error  ERROR Traceback (most recent call last):
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\starlette\routing.py", line 671, in lifespan
    async with self.lifespan_context(app):
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\starlette\routing.py", line 566, in __aenter__
    await self._router.startup()
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\starlette\routing.py", line 648, in startup
    await handler()
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\fastapi_mqtt\fastmqtt.py", line 232, in startup
    await self.connection()
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\fastapi_mqtt\fastmqtt.py", line 130, in connection
    await self.client.connect(
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\gmqtt\client.py", line 225, in connect
    self._connection = await self._create_connection(
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\gmqtt\client.py", line 241, in _create_connection
    connection = await MQTTConnection.create_connection(host, port, ssl, clean_session, keepalive)
  File "C:\Repositories\sandbox\fastapi_mqtt\.venv\lib\site-packages\gmqtt\mqtt\connection.py", line 27, in create_connection
    transport, protocol = await loop.create_connection(MQTTProtocol, host, port, ssl=ssl)
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 1061, in create_connection
    raise exceptions[0]
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 1041, in create_connection
    sock = await self._connect_sock(
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 955, in _connect_sock
    await self.sock_connect(sock, address)
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 702, in sock_connect
    return await self._proactor.connect(sock, address)
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\windows_events.py", line 817, in _poll
    value = callback(transferred, key, ov)
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python39\lib\asyncio\windows_events.py", line 604, in finish_connect
    ov.getresult()
ConnectionRefusedError: [WinError 1225] The remote computer refused the network connection
```

### Request

Are there any examples you can point to that get around this problem? Or will it require some changes to the `fastapi-mqtt` library itself to support transient MQTT broker availability?

### Background

My use case for `fastapi-mqtt` is for writing applications that leverage IoT for inter-process communication; we typically have small groups of applications, often all running on the same host, that rely on a custom in-house MQTT broker service. The MQTT broker service could be shut down, such as when deploying updates, and it would be nice if the separate apps (leveraging `fastapi-mqtt` ) could start up while the broker is down and continue trying to connect until the broker is back up.

TL;DR: I'd love to be able to do say "just keep trying to reconnect to the MQTT broker if you can't establish connection immediately on startup".

**Comment by plundeen** [comment]
Following up: I realize this is probably more of a request on the underlying `gmqtt` library, so I will migrate the request to that repo.

I also managed to come up with a local workaround for now: I'm just including an initialization step in my application that polls the expected broker address/port (via `socket.socket.connect_ex`) until it sees that the broker is up and listening on that port, and _then_ it will attempt the MQTT connection.
